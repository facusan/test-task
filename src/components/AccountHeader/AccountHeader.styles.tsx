import styled from "styled-components";

export const AccountHeaderWrapper = styled.div`
  display: flex;
  height: 80px;
  flex-wrap: wrap;
`;
export const AccountImgWrapper = styled.div`
  padding: 10px;
`;
export const AccountDataWrapper = styled.div`
  display: flex;
  flex-direction: column;
  padding: 10px 2px;
`;
export const AccountNameWrapper = styled.div`
  font-weight: bold;
  font-family: sans-serif;
  margin-bottom: 8px;
  font-size: large;
`;
export const AccountAddressWrapper = styled.div`
  color: lightslategrey;
  font-family: sans-serif;
`;
