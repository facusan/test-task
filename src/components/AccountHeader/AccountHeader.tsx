import {
  AccountAddressWrapper,
  AccountDataWrapper,
  AccountHeaderWrapper,
  AccountImgWrapper,
  AccountNameWrapper,
} from "./AccountHeader.styles";

interface AccountHeaderProps {
  accountName: string;
  accountAddress: string;
  accountImg: string;
}
export default function AccountHeader({
  accountAddress,
  accountName,
  accountImg,
}: AccountHeaderProps) {
  const getShortAddress = () => {
    return `${accountAddress.slice(0, 5)}...${accountAddress.slice(
      accountAddress.length - 4,
      accountAddress.length
    )}`;
  };
  return (
    <AccountHeaderWrapper>
      <AccountImgWrapper>
        <img
          style={{ width: "50px" }}
          src={accountImg}
          alt="account avatar"
        ></img>
      </AccountImgWrapper>
      <AccountDataWrapper>
        <AccountNameWrapper>{accountName}</AccountNameWrapper>
        <AccountAddressWrapper>{getShortAddress()}</AccountAddressWrapper>
      </AccountDataWrapper>
    </AccountHeaderWrapper>
  );
}
