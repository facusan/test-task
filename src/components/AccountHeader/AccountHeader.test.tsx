import renderer from "react-test-renderer";
import AccountHeader from "./AccountHeader";

it("renders AccountHeader correctly", () => {
  const component = renderer.create(
    <AccountHeader
      accountAddress="0x3B375486d5fB0D0fdf94a0FF66092072996Bd1ab"
      accountImg="avatar.png"
      accountName="Account 1"
    ></AccountHeader>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
