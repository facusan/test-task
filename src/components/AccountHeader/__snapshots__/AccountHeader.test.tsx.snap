// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AccountHeader correctly 1`] = `
<div
  className="sc-bczRLJ yJABb"
>
  <div
    className="sc-gsnTZi lEtQN"
  >
    <img
      alt="account avatar"
      src="avatar.png"
      style={
        Object {
          "width": "50px",
        }
      }
    />
  </div>
  <div
    className="sc-dkzDqf GaipU"
  >
    <div
      className="sc-hKMtZM fdRjPi"
    >
      Account 1
    </div>
    <div
      className="sc-eCYdqJ fpmrSv"
    >
      0x3B3...d1ab
    </div>
  </div>
</div>
`;
